packages/flutter/lib/src/widgets/icon.dart

```diff
@@ -122,6 +133,17 @@ class Icon extends StatelessWidget {
     );
   }
 
+  /// Wraps the widget with a Semantics widget if [semanticLabel] is set.
+  Widget _wrapWithSemantics(Widget widget) {
+    if (semanticLabel == null)
```

Comment:
We should always include the Semantics widget. Otherwise, if the semanticLabel is toggled from null to non-null, the entire subtree will be thrown away and rebuilt.